src/components/pdf-recommender/PDFRecommender.jsx

```diff
@@ -0,0 +1,29 @@
+import React, { useEffect, useState } from 'react'
+import load from 'little-loader'
+
+import './PDFRecommender.scss'
+
+const PDFRecommender = ({ containerWidth }) => {
+  const [recommenderLoaded, setRecommenderLoaded] = useState(false)
+
+  useEffect(() => {
+    load('https://core.ac.uk/recommender/embed.js', () => {
+      setRecommenderLoaded(true)
+    })
+  }, [])
+
+  return (
+    <div
+      className="pdf-recommender"
+      style={{
+        width: containerWidth,
+        display: recommenderLoaded && containerWidth ? 'visible' : 'hidden',
+      }}
+    >
+      <h3>Related Articles</h3>
+      <div id="coreRecommenderOutput" />
```

Comment:
I see now why it's not so simple.

However, you might use a [MutationObserver](https://developer.mozilla.org/en/docs/Web/API/MutationObserver) and destruct it after the first callback.

Reply:
I don't think it will help in any way. Would you mind the show me some example? 

The reason for using `little-loader` is that that the `<div id="coreRecommenderOutput" />` has to exist when recommender is loaded. There is no much configuration for recommender right now. The recommender is executed right after it is loaded. 

Even if I managed to insert that `div` before recommender is loaded there is a second problem - metadata. The `<meta>` tags are inserted dynamically and the recommendation wouldn't work properly. Since meta tags could be inserted after recommended is executed.